specifyweb/specify/calculated_fields.py

```diff
@@ -57,10 +65,43 @@ def calculate_extra_fields(obj, data: Dict[str, Any]) -> Dict[str, Any]:
         ).exists()
 
     elif isinstance(obj, get_model("Collectionobject")):
-        preparations = obj.preparations.all()
-        totalCountAmt = calculate_quantity(preparations, "countamt")
+        preparations = Preparation.objects.filter(collectionobject_id=obj.id)
+        totalCountAmt = preparations.aggregate(total=Coalesce(Sum('countamt'), Value(0)))['total']
+
+        # Calculate the sums for all preparations at once
+        giftpreparation_sums = GiftPreparation.objects.filter(
+            preparation__in=preparations
+        ).values('preparation').annotate(total=Coalesce(Sum('quantity'), Value(0)))
+
+        exchangeoutprep_sums = ExchangeOutPrep.objects.filter(
+            preparation__in=preparations
+        ).values('preparation').annotate(total=Coalesce(Sum('quantity'), Value(0)))
+
+        disposalpreparation_sums = DisposalPreparation.objects.filter(
+            preparation__in=preparations
+        ).values('preparation').annotate(total=Coalesce(Sum('quantity'), Value(0)))
+
+        # Convert the sums to dictionaries for faster access
+        giftpreparation_sums = {item['preparation']: item['total'] for item in giftpreparation_sums}
+        exchangeoutprep_sums = {item['preparation']: item['total'] for item in exchangeoutprep_sums}
+        disposalpreparation_sums = {item['preparation']: item['total'] for item in disposalpreparation_sums}
```

Comment:
For readability and maintainability reasons, it is better to use Django ORM code wherever possible.  It's also just best practice for Django applications.  There would need to be significant performance reasons to do otherwise, and if there were, then we should try to use SQL alchemy instead of hard-coded SQL.

Reply:
awesome! just add a note for testers to test it on herb-rbge. they have some big loans I believe.